apps/server/src/v1/statusReportUpdate.ts

```diff
@@ -37,10 +37,15 @@ export const statusUpdateSchema = z.object({
   }),
   id: z.coerce.string().openapi({ description: "The id of the update" }),
   date: z
-    .preprocess((val) => String(val), z.string())
-    .openapi({
-      description: "The date of the update in ISO 8601 format",
-    }),
+  .preprocess((val) => {
+    if (val) {
+      return new Date(String(val)).toISOString();
+    }
+    return new Date().toISOString();
+  }, z.string())
```

Comment:
We should extract that to 
 
```
const isoDate = z
  .preprocess((val) => {
    if (val) {
      return new Date(String(val)).toISOString();
    }
    return new Date().toISOString();
  }, z.string())
  ```
  Thus if we have a single source of truth

Reply:
should I store it in some other file , as its used in many places ??